### DISCUSSION TITLE
Change the plotting front end

### DISCUSSION THREAD

**Discussion by Hardcore-fs** [Chia-Network/chia-blockchain#5834] [Ideas]
1. Change the plotting front end so that we can have predefined groups (plotting area groups)
of " plotting temp" & "plotting final store", instead of the endless nonsense of having to navigate directories to find the directory

2. allow "plotting job groups" to be associated with these plotting area groups
 so that proper queues can be built.

3. allow a "queued plot" to be changed to a "do it now" plot .
4. improve the plotting logic.
currently if i have 1 job allocated to a HDD & 1 job allocated to an SSD
then queue a new job to the SSD directory (space issue) when the HDD job finishes it attempts to start another SSD job!!!!

**Comment by room101-dev** [comment]
> 1. Change the plotting front end so that we can have predefined groups (plotting area groups)
>    of " plotting temp" & "plotting final store", instead of the endless nonsense of having to navigate directories to find the directory
> 2. allow "plotting job groups" to be associated with these plotting area groups
>    so that proper queues can be built.
> 3. allow a "queued plot" to be changed to a "do it now" plot .
> 4. improve the plotting logic.
>    currently if i have 1 job allocated to a HDD & 1 job allocated to an SSD
>    then queue a new job to the SSD directory (space issue) when the HDD job finishes it attempts to start another SSD job!!!!

Chia make/create plot

"Chia Plot create -t /mnt/fast-nvme/tmp1 -2 /mnt-ssd/tmp -d /mnt/Plots .. -n (#plots) ."

That's it, that's the engine that make plots on chai<env>.

Why not just use the cmd-line? The GUI is a faux reality

The chia-plot-create process is what it is -t, -2, -d; there is nothing else to it
The GUI is for morons that can't handle the cmd-line

Now I understand GUI for wallets, because too many morons lost their money, so the GUI is for child-proofing

But what your asking for is an abstraction for something that doesn't exist

For each plot process, there are switches

-t, where the work is to be done, almost always an NVME drive +1TB ( an m.2 drive, of enterprise grade )
-2, an intermediate storage site, for a completed plot to offload -t at stage 3-4 ( almost always a sata-ssd)
-d a target always an hdd for permanent storage of plot
-n how many plots for this task to make
-r how many threads for this task ( > 2 has no meaning, as chia-pos has not implemented parallel processing )

So that's it, now what your asking for is how to hide reality. ALL work is done at the cmdline,, all tools are developed at the cmd-line

THe GUI is just baby diapers so newbies don't shit their pants.

...

There is no such thing as a 'queued plot' that is an abtstraction in your mind, most of us have a batch/bash file that does

The "make-plot.sh"(.bat) queue, invoke everytime you want to start a new batch of 10 plots, here 4x means that -d needs to have 5TB of free space. That -t needs 1TB of free space, that -2 needs 4 * 108GB  of free space

**_wait 0
chia plot create ... -t tmp1 
wait 30m
chia plot create ... -t tmp2
wait 30m
chia plot create ... -t tmp3 
wait 30m
chia plot create ... -t tmp4_** 

This is your 'queue' you run the batch on the completion say ten plots, what this does is stagger here by 30m, which is enough time in stage 4 (20min) to write the -d file, and in stage 1 30 min is enough time to complete the first, and then the second, you don't want all the plot threads trying to do the same thing at once, or the everything will grind to a halt.
PRINT WHAT I HAVE WRITTEN ABOVE AND MEMORIZE
QUIT USING THE GUI
